test/components/views/rooms/RoomListHeader-test.tsx

```diff
@@ -28,12 +30,97 @@ import DMRoomMap from "../../../../src/utils/DMRoomMap";
 import MatrixClientContext from "../../../../src/contexts/MatrixClientContext";
 import SettingsStore from "../../../../src/settings/SettingsStore";
 import { SettingLevel } from "../../../../src/settings/SettingLevel";
+import { shouldShowComponent } from '../../../../src/customisations/helpers/UIComponents';
+import { UIComponent } from '../../../../src/settings/UIFeature';
+
+jest.mock('../../../../src/customisations/helpers/UIComponents', () => ({
+    shouldShowComponent: jest.fn(),
+}));
+
+const blockUIComponent = component => {
+    mocked(shouldShowComponent).mockImplementation(feature => feature !== component);
+};
+
+const setupSpace = (client) => {
+    const testSpace = mkSpace(client, "!space:server");
+    testSpace.name = "Test Space";
+    client.getRoom = () => testSpace;
+    return testSpace;
+};
+
+const setupMainMenu = async (client, testSpace) => {
+    const getUserIdForRoomId = jest.fn();
+    const getDMRoomsForUserId = jest.fn();
+    // @ts-ignore
+    DMRoomMap.sharedInstance = { getUserIdForRoomId, getDMRoomsForUserId };
```

Comment:
I believe you can replace all of this with `DMRoomMap.makeShared();`. That's what our other tests do, at least.

Reply:
`DMRoomMap.makeShared();` is using in own concrete implementation and here we would like to use our mocked implementation I have used 
```
const dmRoomMap = {
            getUserIdForRoomId: jest.fn(),
            getDMRoomsForUserId: jest.fn(),
} as unknown as DMRoomMap;
DMRoomMap.setShared(dmRoomMap);
```

It was performed in some others tests, I hope this is ok